src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,122 @@
 package core.basesyntax;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
 
 public class ArrayList<T> implements List<T> {
+    private Object[] arrayList;
+    private int size;
+    private final int defaultSize = 10;
+
+    public ArrayList() {
+        arrayList = new Object[defaultSize];
+        size = 0;
+    }
+
+    public void ensureCapacity() {
+        if (size == arrayList.length) {
+            arrayList = Arrays.copyOf(arrayList, arrayList.length + (arrayList.length >> 1));
```

Comment:
The use of `Arrays.copyOf` is against the checklist's guideline: 'Don't use Objects, Arrays, or any other util class.' Consider implementing the array copying logic manually or using `System.arraycopy()` as suggested in the checklist .